apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/PgNumeric.cs

```diff
@@ -431,15 +465,70 @@ public static PgNumeric FromDecimal(decimal value)
         /// <returns>The converted value.</returns>
         public decimal ToDecimal(LossOfPrecisionHandling lossOfPrecisionHandling)
         {
-            // FIXME: Awful performance!
-            string text = ToString();
-            // Handles overflow.
-            decimal dec = Decimal.Parse(text, CultureInfo.InvariantCulture);
-            if (lossOfPrecisionHandling == LossOfPrecisionHandling.Throw && this != dec)
+            if (_notANumber)
             {
-                throw new InvalidOperationException("Conversion would lose information");
+                throw new OverflowException($"NaN cannot be represented as {typeof(decimal).FullName}");
```

Comment:
Is this existing behavior for NaN? I'm not sure whether OverflowException is suitable here - I think I'd use InvalidOperationException. But if this is the existing behavior, obviously we should keep it.